pandas/io/pytables.py

```diff
@@ -331,11 +331,20 @@ def read_hdf(path_or_buf, key=None, **kwargs):
 
     try:
         if key is None:
-            keys = store.keys()
-            if len(keys) != 1:
-                raise ValueError('key must be provided when HDF file contains '
-                                 'multiple datasets.')
-            key = keys[0]
+            groups = store.groups()
+            if len(groups) == 0:
+                raise ValueError('No dataset in HDF file.')
```

Comment:
I think this is ok. The user is explicity trying to do something which doesn't work. Its the same type of error when you have multiple keys, so its consistent. (and ok)